### ISSUE TITLE
Regression calling childNodes

### ISSUE THREAD

**Issue by jnthn** [libxml-raku/LibXML-raku#47]
The following code worked fine on `LibXML:ver<0.4.0>`:

```
use LibXML::Document;
my LibXML::Document $backing .= new: :version('1.0'), :enc('UTF-8');
my LibXML::Element $root = $backing.createElementNS(
        'http://schemas.openxmlformats.org/spreadsheetml/2006/main',
        'worksheet');
$root.addNamespace('http://schemas.openxmlformats.org/officeDocument/2006/relationships', 'r');
$backing.setDocumentElement($root);
my LibXML::Element $sheetData = $backing.createElement('sheetData');
$root.add($sheetData);
my @node-list := $backing.documentElement.childNodes.list;
```

On the latest release (`LibXML:ver<0.5.9>`) it dies with:

```
You cannot create an instance of this type (List)
  in sub iterate-list at /home/jnthn/dev/MoarVM/install/share/perl6/site/sources/171A518D00BB5F69960BE34FBB87BB2B8405CDF8 (LibXML::Node) line 728
  in method childNodes at /home/jnthn/dev/MoarVM/install/share/perl6/site/sources/171A518D00BB5F69960BE34FBB87BB2B8405CDF8 (LibXML::Node) line 375
  in block <unit> at golf.raku line 10
```

(This is golfed down from investigating a report that `Spreadsheet::XLSX` had somehow broken.)

**Comment by dwarring** [comment]
Fixed in LibXML 0.5.0 uploaded to CPAN

**Comment by zaucker** [comment]
> Fixed in LibXML 0.5.10 uploaded to CPAN

Great. Thanks for the quick fix!

**Comment by dwarring** [comment]
Yeah, the offending code was a supposed optimization introduced in LibXML 0.5.0. But when I run the benchmarks `etc/benchmark.raku`, I can see no difference anyway.
